lib/dai-plugin-mcd/src/CdpManager.js

```diff
@@ -123,6 +123,26 @@ export default class CdpManager extends LocalService {
     return this.proxyActions[method](...args);
   }
 
+  @tracksTransactionsWithOptions({ numArguments: 5 })
```

Comment:
note that the value of `numArguments` here and below is incorrect

Reply:
Referencing this [comment](https://github.com/makerdao/dai.js/blob/9c88d297c9182dd10b9342559b2389c98c10d2b5/lib/dai-plugin-mcd/src/utils/tracksTransactions.js#L27) in `trackTransactions`, I may be misunderstanding, but `wipe` takes 4 args, `managerAddress, `daiAdapterAddress`, `cdpId` and `wipeAmount`. Including the promise object stated that would be 5?
Similarily, `lockGem` will take 5 but `lockEth` will take 4, these could be split if necessary to ensure the correct number of arguments is passed but we are not doing this in for `lockAndDraw` which follows the same logic of abstracting  `Eth` and `Gem` contract functions.

Reply:
numArguments should correspond to the number of arguments for the function that is being decorated, so in this case `id, ilk, lockAmount, { promise }`

Reply:
Actually in this case we can use `tracksTransactions` instead of `tracksTransactionsWithOptions` because none of the arguments is optional. For `wipe` the second argument is optional so we have to set `numArguments` to 3.

Actually, wait. The second argument to `wipe` shouldn’t be optional. It’d be meaningless to call that without a valid second argument.

Reply:
:+1: